core/src/main/java/edu/ucr/cs/riple/core/Annotator.java

```diff
@@ -216,4 +224,71 @@ private ImmutableSet<Report> processTriggeredFixes(
     // Result of the iteration analysis.
     return explorer.explore();
   }
+
+  /**
+   * Resolves all remaining errors in target module by following steps below:
+   *
+   * <ul>
+   *   <li>Enclosing method of triggered errors will be marked with {@code @NullUnmarked}
+   *       annotation.
+   *   <li>Uninitialized fields (inline or by constructor) will be annotated as
+   *       {@code @SuppressWarnings("NullAway.init")}.
+   *   <li>Explicit {@code Nullable} assignments to fields will be annotated as
+   *       {@code @SuppressWarnings("NullAway")}.
+   * </ul>
+   *
+   * @param fieldDeclarationAnalysis Field declaration analysis.
+   * @param tree Method Declaration analysis.
+   */
+  private void forceResolveRemainingErrors(
+      FieldDeclarationAnalysis fieldDeclarationAnalysis, MethodDeclarationTree tree) {
+    // Collect regions with remaining errors.
+    Utility.buildTarget(config);
+    List<Error> remainingErrors = Utility.readErrorsFromOutputDirectory(config.target);
+    Set<Fix> remainingFixes =
+        Utility.readFixesFromOutputDirectory(
+            config.target, Fix.factory(config, fieldDeclarationAnalysis));
+
+    // Collect all regions for NullUnmarked.
+    Set<AddAnnotation> nullUnMarkedAnnotations =
+        remainingErrors.stream()
+            // filter non-method regions.
+            .filter(error -> !error.encMethod().equals("null"))
+            // find the corresponding method nodes.
+            .map(error -> tree.findNode(error.encMethod(), error.encClass()).location)
+            // impossible, just sanity check or future nullness checker hints
+            .filter(Objects::nonNull)
+            .map(location -> new AddAnnotation(location, config.nullUnMarkedAnnotation))
```

Comment:
Don't we need a way to pass the reason this error wasn't/couldn't be fixed here somehow? (To be added in a comment, not a string, but either way it would be good to have)

Reply:
I didnt fully understand what to do here, the remaining errors are mostly due their corresponding fixes get rejected (error in downstream or effect more than 0) or no corresponding fix exists.

Reply:
So, what I mean here is, since we can't actually have:

```
@NullUnmarked("Some text")
public Object foo {
```

But we want to be able to inject some sort of explanation for why we are unmarking. I think we need either:

```
// Explanation: Adding @Nullable here causes 102 errors in downstream dependencies
@NullUnmarked
public Object foo {
```

Or, even *better*:

```
// [NULLAWAY_ANNOTATOR_ISSUE_{UUID}]
@NullUnmarked
public Object foo {
```

with a file (**issues.tsv**?) on the side mapping `NULLAWAY_ANNOTATOR_ISSUE_{UUID}` to a description of the reason the auto-annotator added the unmarked annotation (the idea here being that a simple search and replace can: a) create a new JIRA task and copy in the reason from `issues.tsv`, b) search and replace the code, changing the corresponding `NULLAWAY_ANNOTATOR_ISSUE_{UUID}` string to a link to the issue.

That said, I think this can be a follow up for this diff, and very possible a post-internship task for either of us, given the timeline.

Reply:
Sure, I can do that as a follow up. We can discuss this before the internship ends on what is best output.